File: problem/Maximum_Words_Can_Type/canBeTypedWords.c
int canBeTypedWords(char *text, char *broken)
{
    int b[26] = {0};
    for (int i = 0; broken[i]; ++i)
        b[broken[i] - 'a'] = 1;
    int count = 0, inWord = 0, ok = 1;
    for (int i = 0;; ++i)
    {
        char c = text[i];
        if (c != '\0' && c != ' ')
        {
            if (!inWord)
            {
                inWord = 1;
                ok = 1;
            }
            if (c >= 'a' && c <= 'z' && b[c - 'a'])
                ok = 0;
        }
        else
        {
            if (inWord)
            {
                if (ok)
                    count++;
                inWord = 0;
            }
            if (c == '\0')
                break;
        }
    }
    return count;
}

File: problem/Maximum_Words_Can_Type/canBeTypedWords.js
/**
 * @param {string} text
 * @param {string} brokenLetters
 * @return {number}
 */
const canBeTypedWords = (text, broken) => {
    const brokenKeys = new Set(broken);
    const words = text.split(" ");
    let count = 0;

    for (const word of words) {
        for (const c of word) {
            if (brokenKeys.has(c)) {
                count++;
                break;
            }
        }
    }

    return words.length - count;
};